### PR TITLE
Update to zip 2.0

### LINKED ISSUE #14: The file name is incorrectly encoded after the file is decompressed
My code is as follows, when I unzip my zip file, the Chinese file name inside it is wrongly encoded

``` rust
use std::io::Cursor;
use std::path::{Path, PathBuf};

pub fn extract_zip(zip_buffer: Vec<u8>, dest: &Path) -> Result<(), String> {
    match zip_extract::extract(
        Cursor::new(zip_buffer),
        dest,
        false,
    ) {
        Ok(_) => Ok(()),
        Err(err) => Err(format!("{}", err))
    }
}
```

below is what the code unzips and how it should actually look like
image
image

### PR DESCRIPTION
**Pull request by DaniPopes**
Plus some cargo/clippy warning fixes.

Closes https://github.com/MCOfficer/zip-extract/pull/31
Fixes https://github.com/MCOfficer/zip-extract/issues/14

### CODE REVIEW

File: Cargo.toml
Comment (MCOfficer) [inline Cargo.toml]:
```suggestion
version = "0.1.3"
```

I agree that it should be released as 0.2.0, but I'll let cargo-release do it after the merge.

**Review by MCOfficer — CHANGES_REQUESTED** [review]
Thank you for the PR, this all seems reasonable. I'd love for the tests to use `diff_dir`, but as long as it's flaky i guess this is the better solution.

**Comment by DaniPopes** [comment]
FYI `dir_diff` wasn't removed, just the `use` statement which was unused.

**Comment by MCOfficer** [comment]
Oh, i misread that in conjunction with the collapsed `assert!`. thanks!

**Pull request merged**
